server/src/apub/mod.rs

```diff
@@ -88,3 +124,35 @@ fn is_apub_id_valid(apub_id: &Url) -> bool {
     None => false,
   }
 }
+
+// TODO Not sure good names for these
+pub trait ToApub<Response> {
+  fn to_apub(&self, conn: &PgConnection) -> Result<Response, Error>;
+}
+
+pub trait FromApub<ApubType> {
+  fn from_apub(apub: &ApubType, conn: &PgConnection) -> Result<Self, Error>
+  where
+    Self: Sized;
+}
```

Comment:
I hate these names, but I added some generics for converting to/from apub. This will be useful once we add other apub objects like comments, likes, etc and have to do similar things with them.

Reply:
Is this another case where associated types might make sense, or will you impl ToApub multiple times for each thing?

Reply:
Ya that's a good point, I'll make that change. There's no situation where I'd need those to have *different* response types. Here's a good thread I found on this : https://stackoverflow.com/questions/32059370/when-is-it-appropriate-to-use-an-associated-type-versus-a-generic-type